pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -224,6 +217,31 @@ func (pv *Provisioner) cloneVirtualMachine(s *SessionContext, cluster *clusterv1
 		return fmt.Errorf("error fetching virtual machine or template properties: %s", err)
 	}
 
+	host, err := src.HostSystem(ctx)
+	if err != nil {
+		return fmt.Errorf("HostSystem failed. err=%s", err)
+	}
+
+	pool, err := s.finder.ResourcePoolOrDefault(ctx, machineConfig.MachineSpec.ResourcePool)
+	if err != nil {
+		klog.Warningf("Failed to find ResourcePool=%s err=%s. Attempting to create it.", machineConfig.MachineSpec.ResourcePool, err)
```

Comment:
You can limit the error case where pool doesn't exist like so:
```go
if _, ok := err.(*find.NotFoundError); !ok {
   return err
}
// ... attempt to create the pool
```

Reply:
Good suggestion... will update.

Reply:
Implemented and testing now